lambdas/pkgpush/src/t4_lambda_pkgpush/__init__.py

```diff
@@ -635,28 +709,48 @@ def package_prefix(event, context):
     setup_user_boto_session_from_default()
 
     pkg = quilt3.Package()
-    pkg.set_dir(".", str(prefix_pk), meta=metadata)
-    # TODO: check while listing objects
+
+    prefix_len = len(prefix_pk.path)
     size_to_hash = 0
-    for i, (_, pkg_entry) in enumerate(pkg.walk()):
-        if i > MAX_FILES_TO_HASH:
+    files_to_hash = 0
+
+    for obj in list_prefix_latest_versions(prefix_pk.bucket, prefix_pk.path):
+        key = obj["Key"]
+        size = obj["Size"]
+        # XXX: disable limits?
```

Comment:
@nl0 
what do you think about this?

Reply:
yeah i think it's ok

Reply:
(or at least we can come up with some more up-to-date heuristics to abort "known negatives" idk)